src/Umbraco.PublishedCache.HybridCache/NotificationHandlers/HybridCacheStartupNotificationHandler.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Umbraco.Cms.Core;
+using Umbraco.Cms.Core.Events;
+using Umbraco.Cms.Core.Notifications;
+using Umbraco.Cms.Core.PublishedCache;
+using Umbraco.Cms.Core.Services;
+
+namespace Umbraco.Cms.Infrastructure.HybridCache.NotificationHandlers;
+
+/// <summary>
+///     Rebuilds the database cache if required when the serializer changes
+/// </summary>
+public class HybridCacheStartupNotificationHandler : INotificationAsyncHandler<UmbracoApplicationStartingNotification>
+{
+    private readonly IDatabaseCacheRebuilder _databaseCacheRebuilder;
+    private readonly IRuntimeState _runtimeState;
+
+    public HybridCacheStartupNotificationHandler(IDatabaseCacheRebuilder databaseCacheRebuilder, IRuntimeState runtimeState)
+    {
+        _databaseCacheRebuilder = databaseCacheRebuilder;
+        _runtimeState = runtimeState;
+    }
+
+    public Task HandleAsync(UmbracoApplicationStartingNotification notification, CancellationToken cancellationToken)
+    {
+        if (_runtimeState.Level == RuntimeLevel.Run)
```

Comment:
I guess this needs to happen if it is just larger then install. Also Update is runnable when maintenance page is disabled

Reply:
I guess we can do `(_runtimeState.Level > RuntimeLevel.Upgrade || (_runtimeState.Level == RuntimeLevel.Upgrade && _globalSettings.ShowMaintenancePageWhenInUpgradeState)`  🤔

Reply:
I thin it's fine to do when runtime level is upgrade no matter what.

I have some issues to move back from JSON to MsgPack.. This seems to execute before the seeding. Maybe we will need to merge the two notification handlers

Reply:
Wierd that it runs before, as seeding should be `Started`, and this is `Starting` 😕 
Maybe it's because of the async notification handler ? 🤔